### ISSUE TITLE
Optimizer Overlap: Follow up features

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#71595]
### 🚀 The feature, motivation and pitch

A process to overlap optimizer + allreduce in DDP is being implemented as described in https://github.com/pytorch/pytorch/issues/67570, although as a prototype feature there is limited support for some use cases. This issue exists to maintain a list of currently missing features and track demand for them:

- [ ] Supporting registering of multiple optimizers, each with possibly different (and possibly overlapping) subsets of DDP model parameters
- [ ] Supporting DDP `no_sync` mode
- [ ] Supporting gradient accumulation
- [ ] Supporting registering a custom communication in addition to overlapped optimizer

If any features (whether they are captured above or not) are needed for your use case with this feature, please ping this issue. Thank you!
